block/fork.go

```diff
@@ -0,0 +1,110 @@
+package block
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/dymensionxyz/dymint/node/events"
+	"github.com/dymensionxyz/dymint/types"
+	uevent "github.com/dymensionxyz/dymint/utils/event"
+)
+
+// MonitorForkUpdate listens to the hub
+func (m *Manager) MonitorForkUpdate(ctx context.Context) error {
+	err := m.checkForkUpdate(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(15 * time.Second)
+	defer ticker.Stop()
+
+	for {
+		select {
+		case <-ctx.Done():
+			return nil
+		case <-ticker.C:
+			if err := m.checkForkUpdate(ctx); err != nil {
+				continue
+			}
+		}
+	}
+}
+
+// checkForkUpdate checks if the hub has a fork update
+func (m *Manager) checkForkUpdate(ctx context.Context) error {
+	rollapp, err := m.SLClient.GetRollapp()
+	if err != nil {
+		return err
+	}
+
+	lastBlock, err := m.Store.LoadBlock(m.State.Height())
+	if err != nil {
+		return err
+	}
+
+	if m.shouldStopNode(rollapp, lastBlock) {
+		err = m.createInstruction(rollapp, lastBlock)
+		if err != nil {
+			return err
+		}
+
+		m.freezeNode(ctx)
+	}
+
+	return nil
+}
+
+func (m *Manager) createInstruction(rollapp *types.Rollapp, block *types.Block) error {
+	info, err := m.SLClient.GetStateInfo(block.Header.Height)
```

Comment:
if the state info is only necessary for the next proposer, there is already implemented GetNextProposer() func